trace-sdk/src/main/java/io/bitrise/trace/data/management/DataManager.java

```diff
@@ -286,20 +305,21 @@ public void startSending(@NonNull final Context context) {
   /**
    * Stops the sending of any pending requests to the server. These request will be dismissed and
    * will not be sent. No future requests will be sent unless {@link #startSending} is called again.
-   *
-   * @param context the Android Context.
    */
-  public void stopSending(@NonNull final Context context) {
```

Comment:
I refactored stopSending a little bit :

Previously - stopSending would create the metricServiceScheduler and traceServiceScheduler if they didn't exist, to then call cancel - then startSending would check if they were null (which they wouldn't ever be).

I've now made it, so that it checks if the metricServiceScheduler or traceServiceScheduler is not null before posting a log (I think this was misleading to people who read the logs) - it then only calls cancel on them if they were created.  This means the creation of objects stays in start which i think makes more sense.

 Does that sound okay?

Reply:
Sure.